### PR TITLE
NIAD-764 Pipeline: Restore project structure, unit test reports, run tests in single stage

### PR DESCRIPTION
**Pull request by mattd-kainos**
* Restore /project structure
* Adds a DISABLE_TEST_CONTAINERS environment variable for integration tests

### CODE REVIEW

**Review by Broom94 — APPROVED** [review]

**Review by LukaszKwasniewski — DISMISSED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -46,31 +47,20 @@ pipeline {
...
-                            sh '''
-                                docker-compose -f docker-compose-integration-tests.yml build
-                                docker-compose -f docker-compose-integration-tests.yml up --exit-code-from integration_tests
-                            '''
-                        }
-                    }
-                    post {
-                        always {
-                            sh "docker-compose -f docker-compose-integration-tests.yml down --rmi=all"
+                            // TODO: How to make this re-use layers built in test stage? Maybe using docker-compose instead of bare docker would help it find them?
```
Comment (mmackiewicz1) [inline Jenkinsfile:58]: Do we really need this comment?

File: service/src/intTest/java/uk/nhs/adaptors/gp2gp/TestContainerUtils.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package uk.nhs.adaptors.gp2gp;
+
+public class TestContainerUtils {
+    public static boolean isTestContainersEnabled() {
+        return !"true".equalsIgnoreCase(System.getenv("DISABLE_TEST_CONTAINERS"));
```
Comment (mmackiewicz1) [inline service/src/intTest/java/uk/nhs/adaptors/gp2gp/TestContainerUtils.java:5]: Could we move these strings to constants?

File: service/src/test/java/uk/nhs/adaptors/gp2gp/TemporaryTest.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package uk.nhs.adaptors.gp2gp;
+
+import org.junit.jupiter.api.Test;
+
+class TemporaryTest {
```
Comment (mmackiewicz1) [inline service/src/test/java/uk/nhs/adaptors/gp2gp/TemporaryTest.java:5]: Please remove this class. It does not do anything.

Comment (mattd-kainos) [reply service/src/test/java/uk/nhs/adaptors/gp2gp/TemporaryTest.java:5]: It is needed to produce a test report in Jenkins

Comment (mmackiewicz1) [reply service/src/test/java/uk/nhs/adaptors/gp2gp/TemporaryTest.java:5]: Ok

File: Jenkinsfile (hunk 1 above)
Comment (mattd-kainos) [reply Jenkinsfile]: removed comment and created tech debt story

**Review by mmackiewicz1 — APPROVED** [review]

**Review by Broom94 — APPROVED** [review]

**Pull request merged**